tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenAggregateTest.cs

```diff
@@ -102,5 +104,22 @@ protected override void SetArgument(string name, BsonValue value)
 
             base.SetArgument(name, value);
         }
+
+        // private methods
+        private void ReplaceCollectionForAssertingtIfRequired(IEnumerable<BsonDocument> pipelineStages)
+        {
+            // assuming that only one _objectMap item will be added for each call
+            foreach (var stage in pipelineStages)
+            {
+                if (stage.TryGetElement("$out", out var outElement))
+                {
+                    _objectMap.Add(outElement.Name, outElement.Value);
```

Comment:
this code is used to change the outcome collection which will be asserted in the last step in a test for out and merge operations.

I think instead of these lines,  it's also possible to update somehow the collectionName field after the operations, for example, we can create a new empty method like `PostOperationsAction` into `JsonDrivenTest` which will be overridden only in the aggregate test and will update a collection name. 
Another solution can be inheriting `JsonDrivenAggregateTest` from a new interface like `IOutcomeCollectionNameChanged` and save the possibly changed outcome collection name somewhere here: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Tests/Specifications/Runner/MongoClientJsonDrivenTestRunnerBase.cs#L231 and then we will be able to update CollectionName value which is used in `AssertOutcome`

Reply:
so, let me know what do you thin about it

Reply:
This change is not needed. The JSON test file has the name of the output collection. We can get it from there.